src/webgl/p5.RendererGL.js

```diff
@@ -1914,33 +1915,36 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
     *  To create a blurry image from the input non blurry img, if it doesn't already exist
     *  Add it to the diffusedTexture map,
     *  Returns the blurry image
-    *  maps a p5.Image used by imageLight() to a p5.Graphics
+    *  maps a p5.Image used by imageLight() to a p5.Framebuffer
    */
   getDiffusedTexture(input) {
     // if one already exists for a given input image
     if (this.diffusedTextures.get(input) != null) {
       return this.diffusedTextures.get(input);
     }
     // if not, only then create one
-    let newGraphic; // maybe switch to framebuffer
+    let newFramebuffer;
     // hardcoded to 200px, because it's going to be blurry and smooth
     let smallWidth = 200;
     let width = smallWidth;
     let height = Math.floor(smallWidth * (input.height / input.width));
-    newGraphic = this._pInst.createGraphics(width, height, constants.WEBGL);
-    // create graphics is like making a new sketch, all functions on main
-    // sketch it would be available on graphics
-    let irradiance = newGraphic.createShader(
-      defaultShaders.imageLightVert,
-      defaultShaders.imageLightDiffusedFrag
-    );
-    newGraphic.shader(irradiance);
-    irradiance.setUniform('environmentMap', input);
-    newGraphic.noStroke();
-    newGraphic.rectMode(newGraphic.CENTER);
-    newGraphic.rect(0, 0, newGraphic.width, newGraphic.height);
-    this.diffusedTextures.set(input, newGraphic);
-    return newGraphic;
+    newFramebuffer = this._pInst.createFramebuffer();
```

Comment:
I think this makes the framebuffer match the size of the main canvas, but we calculate a different size for it above. We can use that size if we do:
```suggestion
    newFramebuffer = this._pInst.createFramebuffer({
      width,
      height,
      density: 1,
    });
```

Reply:
Great suggestion, it was an oversight by me. I have changed it in specularShader too as we were matching the size of the main canvas before the loop. Although, when we are inside the loop, we are resizing it again.